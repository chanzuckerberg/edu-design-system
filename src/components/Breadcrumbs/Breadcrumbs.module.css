@import '../../design-tokens/mixins.css';

/*------------------------------------*\
    # BREADCRUMBS
\*------------------------------------*/

/**
 * Breadcrumbs are a navigational component
 *
 * The outer wrapper for the breadcrumbs is a <nav> element.
 */

/**
 * Breadcrumbs list
 */
.breadcrumbs__list {
  list-style: none;
  display: flex;
}

/*------------------------------------*\
    # BREADCRUMBS ITEM
\*------------------------------------*/

/**
 * Breadcrumbs list item.
 */
.breadcrumbs__item {
  max-width: 100%;
  /* Required for the Menu to absolutely position relative to this container. */
  position: relative;

  /* Hides all breadcrumbs except the last breadcrumb in smaller breakpoints. */
  display: none;
  flex-shrink: 0;

  &:last-of-type {
    display: flex;
    /* Truncate last breadcrumb in smaller breakpoints */
    flex: 1 0 0%;
  }

  min-width: 0;

  align-items: center;

  @media all and (min-width: $eds-bp-md) {
    /* Display breadcrumbs in larger breakpoints. */
    display: flex;

    &:last-of-type {
      /* Truncate last breadcrumb in smaller breakpoints */
      flex: 0 0 auto;
    }
  }
}

/**
 * Back variant of the breadcrumbs list item.
 */
.breadcrumbs__item-back {
  display: flex;
  margin-right: calc(var(--eds-spacing-size-1-and-half) / 16 * 1rem);

  @media all and (min-width: $eds-bp-md) {
    /* Hidden for larger breakpoints. */
    display: none;
  }
}

.breadcrumbs__item-back>.breadcrumbs__separator {
  /* Hide the separator for the back variant. */
  display: none;
}

/**
 * Ellipsis variant of the breadcrumbs list item.
 */
.breadcrumbs__ellipsis {
  min-height: unset;
  min-width: unset;

  border: none;
  border-radius: unset;
  background-color: transparent;
}

.breadcrumbs__ellipsis:hover {
  background-color: transparent;
}

/**
 * Breadcrumbs link.
 */
.breadcrumbs__link {
  color: var(--eds-theme-color-text-neutral-subtle);
  text-decoration: none;

  &:last-of-type {
    /* Truncate last link with ellipsis. */
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  &:hover,
  &:focus-visible {
    text-decoration: underline;
  }

  &:focus-visible {
    @mixin focus;
  }

  @supports not selector(:focus-visible) {

    &:hover,
    &:focus {
      text-decoration: underline;
    }

    &:focus {
      @mixin focus;
    }
  }
}

/**
 * Breadcrumbs Icon - a separator between breadcrumb links.
 */
.breadcrumbs__separator {
  color: var(--eds-theme-color-text-utility-default-primary);
  margin-left: calc(var(--eds-spacing-size-1) / 16 * 1rem);
  margin-right: calc(var(--eds-spacing-size-1) / 16 * 1rem);
  cursor: default;
}

/**
 * Last breadcrumbs item icon.
 */
.breadcrumbs__item:last-child .breadcrumbs__separator.breadcrumbs__separator {
  /*  A separator shouldn't be displayed after last link. */
  display: none;
}

/**
 * Breadcrumbs Back Icon.
 */
.breadcrumbs__back-icon {
  color: var(--eds-theme-color-icon-utility-default-primary);
  /* Transform over height due to icon being placed inside <a>. */
  transform: scale(1.5);
  position: relative;
  bottom: 0.125rem;
}