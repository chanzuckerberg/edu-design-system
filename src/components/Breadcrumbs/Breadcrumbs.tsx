import clsx from 'clsx';
import debounce from 'lodash.debounce';
import React, { type ReactNode } from 'react';
import { useUID } from 'react-uid';
import styles from './Breadcrumbs.module.css';
import { flattenReactChildren } from '../../util/flattenReactChildren';
import BreadcrumbsItem from '../BreadcrumbsItem';
import DropdownMenuItem from '../DropdownMenuItem';

type Props = {
  /**
   * Child node(s) that can be nested inside component
   */
  children?: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  /**
   * aria-label for `nav` element to describe Breadcrumbs navigation to screen readers
   */
  'aria-label'?: string;

  /**
   * HTML id for the component
   */
  id?: string;
};

/**
 * `import {Breadcrumbs} from "@chanzuckerberg/eds";`
 *
 * List of Breadcrumb components showing the user where they are in the system and allow them
 * to navigate to parent pages.
 */
export const Breadcrumbs = ({
  className,
  children,
  id,
  'aria-label': ariaLabel = 'breadcrumbs links',
  ...other
}: Props) => {
  const [shouldTruncate, setShouldTruncate] = React.useState(false);

  const ref = React.useRef<HTMLUListElement>(null);

  /**
   * Needs useLayoutEffect over useEffect since it needs to be run before paint.
   * TODO: with React 18, might be able to use useEffect https://github.com/reactjs/reactjs.org/blob/d14cbdca2445cd676526c4c52e1e106342ff7bb3/content/docs/hooks-reference.md?plain=1#L155
   */
  React.useLayoutEffect(() => {
    const updateShouldTruncate = () => {
      const willOverflow = ref.current
        ? ref.current.clientWidth < ref.current.scrollWidth
        : false;

      setShouldTruncate(willOverflow);
    };

    const debouncedUpdateShouldTruncate = debounce(updateShouldTruncate, 200);

    updateShouldTruncate();
    window.addEventListener('resize', debouncedUpdateShouldTruncate);
    return () => {
      window.removeEventListener('resize', debouncedUpdateShouldTruncate);
    };
  }, []);

  /**
   * Flattens breadcrumb items that may be wrapped in React.Fragment into an array so they can be manipulated easily.
   */
  const breadcrumbsItems = flattenBreadcrumbsItems(children);

  /**
   * Finds the second to last breadcrumb item for mobile Breadcrumbs back icon.
   */
  const backBreadCrumb =
    breadcrumbsItems.length > 1
      ? React.cloneElement(
          breadcrumbsItems[
            breadcrumbsItems.length - 2
          ] as React.ReactElement<any>,
          {
            variant: 'back',
          },
        )
      : null;

  /**
   * Finds all the breadcrumb items between the first and last breadcrumb items so they can be placed in the dropdown.
   */
  const dropdownMenuItems = breadcrumbsItems
    .slice(1, breadcrumbsItems.length - 1)
    .map((breadcrumbItem, index) => {
      const menuItem = breadcrumbItem as JSX.Element;
      return (
        <DropdownMenuItem
          href={menuItem.props.href}
          // FIXME
          // eslint-disable-next-line react/no-array-index-key
          key={`breadcrumb-menu-item-${index}`}
        >
          {menuItem.props.text}
        </DropdownMenuItem>
      );
    });

  const generatedId = useUID();
  const breadcrumbsId = id || generatedId;

  const componentClassName = clsx(styles['breadcrumbs'], className);
  return (
    <nav
      aria-label={ariaLabel}
      className={componentClassName}
      id={breadcrumbsId}
      {...other}
    >
      <ul className={styles['breadcrumbs__list']} ref={ref}>
        {/**
         * Back icon breadcrumb always exists, just hidden via css depending on breakpoint to increase performance
         */}
        {backBreadCrumb}
        {/**
         * The ellipsis breadcrumb with dropdown only exists if there would be overflow and there are 3 or more breadcrumb items.
         */}
        {shouldTruncate && breadcrumbsItems.length > 2 ? (
          <>
            {breadcrumbsItems[0]}
            <BreadcrumbsItem
              dropdownMenuItems={dropdownMenuItems}
              href={null}
              variant="collapsed"
            />
            {breadcrumbsItems[breadcrumbsItems.length - 1]}
          </>
        ) : (
          /**
           * If the above conditions aren't met, display all breadcrumbs.
           */
          breadcrumbsItems
        )}
      </ul>
    </nav>
  );
};

const flattenBreadcrumbsItems = (children: React.ReactNode) => {
  const flattenedChildren = flattenReactChildren(children);
  /**
   * Throws error if children are not BreadcrumbsItems
   */
  const shouldThrowError = (flattenedChildren as JSX.Element[]).some(
    (child) => {
      if (child.type === BreadcrumbsItem || child.type === Breadcrumbs.Item) {
        return false;
      }
      return true;
    },
  );
  if (process.env.NODE_ENV !== 'production' && shouldThrowError) {
    throw 'Only <Breadcrumbs.Item>, <BreadcrumbsItem>, or React.Fragment of aforementioned components allowed';
  }
  return flattenedChildren;
};

Breadcrumbs.Item = BreadcrumbsItem;
