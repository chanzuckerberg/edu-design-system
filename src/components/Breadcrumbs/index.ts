export { Breadcrumbs as default } from './Breadcrumbs';
