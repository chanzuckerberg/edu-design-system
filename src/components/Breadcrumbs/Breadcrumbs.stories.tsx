import type {StoryObj, Meta} from '@storybook/react';
import {within} from '@storybook/testing-library';
import React from 'react';

import {Breadcrumbs} from './Breadcrumbs';

export default {
  title: 'Molecules/Navigation/Breadcrumbs',
  component: Breadcrumbs,
  subcomponents: {'Breadcrumbs.Item': Breadcrumbs.Item},
  args: {
    children: (
      <>
        <Breadcrumbs.Item href="#" text="Home" />
        <Breadcrumbs.Item href="#" text="Child" />
        <Breadcrumbs.Item href="#" text="Grandchild" />
      </>
    ),
  },
  decorators: [
    (Story) => (
      <div
        style={{
          margin: '0.5rem',
        }}
      >
        <Story />
      </div>
    ),
  ],
} as Meta<Args>;

type Args = React.ComponentProps<typeof Breadcrumbs>;

export const Default: StoryObj<Args> = {};

export const OneCrumb: StoryObj<Args> = {
  args: {
    children: <Breadcrumbs.Item href="#" text="Breadcrumb 1" />,
  },
};

export const TwoCrumbs: StoryObj<Args> = {
  args: {
    children: (
      <>
        <Breadcrumbs.Item href="#" text="Breadcrumb 1" />
        <Breadcrumbs.Item href="#" text="Breadcrumb 2" />
      </>
    ),
  },
};

export const LongList: StoryObj<Args> = {
  args: {
    children: (
      <>
        <Breadcrumbs.Item href="#" text="Parent" />
        <Breadcrumbs.Item href="#" text="Breadcrumb 1" />
        <Breadcrumbs.Item href="#" text="Breadcrumb 2" />
        <Breadcrumbs.Item href="#" text="Breadcrumb 3" />
        <Breadcrumbs.Item href="#" text="Breadcrumb 4" />
        <Breadcrumbs.Item href="#" text="Breadcrumb 5" />
        <Breadcrumbs.Item href="#" text="Breadcrumb 6" />
        <Breadcrumbs.Item href="#" text="Breadcrumb 7" />
        <Breadcrumbs.Item href="#" text="Breadcrumb 8" />
        <Breadcrumbs.Item href="#" text="Breadcrumb 9" />
        <Breadcrumbs.Item href="#" text="Breadcrumb 10" />
      </>
    ),
  },
};

export const LongText: StoryObj<Args> = {
  args: {
    children: (
      <>
        <Breadcrumbs.Item href="#" text="Home" />
        <Breadcrumbs.Item href="#" text="Breadcrumb 1" />
        <Breadcrumbs.Item
          href="#"
          text="Breadcrumb 2 Lorem ipsum dolor sit amet, no overflow is two lines at 320px"
        />
        <Breadcrumbs.Item
          href="#"
          text="Breadcrumb 3 Lorem ipsum dolor sit amet, consectetur adipiscing elit, no overflow is 3 lines at 320px"
        />
      </>
    ),
  },
  parameters: {
    chromatic: {viewports: [414, 768, 1366]},
  },
};

/**
 * Mostly for visual regression testing.
 */
export const LongTextMenu: StoryObj<Args> = {
  args: {
    ...LongText.args,
  },
  parameters: {
    viewport: {
      defaultViewport: 'tablet',
    },
    chromatic: {viewports: [834]},
    axe: {
      skip: true,
    },
    snapshot: {
      skip: true,
    },
  },
  play: async ({canvasElement}) => {
    const canvas = within(canvasElement);
    const dropdownMenuTrigger = await canvas.findByRole('button');
    dropdownMenuTrigger.click();
  },
};
