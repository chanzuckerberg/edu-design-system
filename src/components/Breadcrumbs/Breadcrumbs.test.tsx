import {generateSnapshots} from '@chanzuckerberg/story-utils';
import {composeStories} from '@storybook/testing-react';
import {render, screen, waitFor} from '@testing-library/react';
import React from 'react';
import * as stories from './Breadcrumbs.stories';

const {LongList} = composeStories(stories);

describe('<Breadcrumbs />', () => {
  generateSnapshots(stories);

  describe('truncation', () => {
    it('truncates when its content overflows', async () => {
      render(<LongList />);
      const list = screen.getByRole('list');

      Object.defineProperty(list, 'clientWidth', {value: 100});
      Object.defineProperty(list, 'scrollWidth', {value: 200});
      expect(list.clientWidth).toBeLessThan(list.scrollWidth);

      window.dispatchEvent(new Event('resize'));

      await waitFor(() => {
        expect(screen.getAllByRole('listitem').length).toEqual(4);
      });
    });

    it('does not truncate when its content does not overflow', async () => {
      render(<LongList />);
      const list = screen.getByRole('list');

      Object.defineProperty(list, 'clientWidth', {value: 200});
      Object.defineProperty(list, 'scrollWidth', {value: 200});
      expect(list.clientWidth).toBeGreaterThanOrEqual(list.scrollWidth);

      window.dispatchEvent(new Event('resize'));

      await waitFor(() => {
        expect(screen.getAllByRole('listitem').length).toEqual(12);
      });
    });
  });
});
