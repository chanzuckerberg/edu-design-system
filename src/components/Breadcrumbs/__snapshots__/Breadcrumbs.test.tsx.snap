// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumbs /> Default story renders snapshot 1`] = `
<div
  style="margin: 0.5rem;"
>
  <nav
    aria-label="breadcrumbs links"
    class="breadcrumbs"
    id="1"
  >
    <ul
      class="breadcrumbs__list"
    >
      <li
        class="breadcrumbs__item breadcrumbs__item-back"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          <svg
            class="icon breadcrumbs__back-icon"
            fill="currentColor"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="2"
            >
              Child
            </title>
            <path
              d="M16 22L6 12L16 2L17.775 3.775L9.55 12L17.775 20.225L16 22Z"
            />
          </svg>
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Home
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Child
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Grandchild
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`<Breadcrumbs /> LongList story renders snapshot 1`] = `
<div
  style="margin: 0.5rem;"
>
  <nav
    aria-label="breadcrumbs links"
    class="breadcrumbs"
    id="6"
  >
    <ul
      class="breadcrumbs__list"
    >
      <li
        class="breadcrumbs__item breadcrumbs__item-back"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          <svg
            class="icon breadcrumbs__back-icon"
            fill="currentColor"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="7"
            >
              Breadcrumb 9
            </title>
            <path
              d="M16 22L6 12L16 2L17.775 3.775L9.55 12L17.775 20.225L16 22Z"
            />
          </svg>
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Parent
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 1
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 2
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 3
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 4
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 5
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 6
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 7
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 8
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 9
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 10
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`<Breadcrumbs /> LongText story renders snapshot 1`] = `
<div
  style="margin: 0.5rem;"
>
  <nav
    aria-label="breadcrumbs links"
    class="breadcrumbs"
    id="8"
  >
    <ul
      class="breadcrumbs__list"
    >
      <li
        class="breadcrumbs__item breadcrumbs__item-back"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          <svg
            class="icon breadcrumbs__back-icon"
            fill="currentColor"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="9"
            >
              Breadcrumb 2 Lorem ipsum dolor sit amet, no overflow is two lines at 320px
            </title>
            <path
              d="M16 22L6 12L16 2L17.775 3.775L9.55 12L17.775 20.225L16 22Z"
            />
          </svg>
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Home
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 1
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 2 Lorem ipsum dolor sit amet, no overflow is two lines at 320px
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 3 Lorem ipsum dolor sit amet, consectetur adipiscing elit, no overflow is 3 lines at 320px
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`<Breadcrumbs /> OneCrumb story renders snapshot 1`] = `
<div
  style="margin: 0.5rem;"
>
  <nav
    aria-label="breadcrumbs links"
    class="breadcrumbs"
    id="3"
  >
    <ul
      class="breadcrumbs__list"
    >
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 1
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`<Breadcrumbs /> TwoCrumbs story renders snapshot 1`] = `
<div
  style="margin: 0.5rem;"
>
  <nav
    aria-label="breadcrumbs links"
    class="breadcrumbs"
    id="4"
  >
    <ul
      class="breadcrumbs__list"
    >
      <li
        class="breadcrumbs__item breadcrumbs__item-back"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          <svg
            class="icon breadcrumbs__back-icon"
            fill="currentColor"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="5"
            >
              Breadcrumb 1
            </title>
            <path
              d="M16 22L6 12L16 2L17.775 3.775L9.55 12L17.775 20.225L16 22Z"
            />
          </svg>
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 1
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__link"
          href="#"
        >
          Breadcrumb 2
        </a>
        <span
          aria-hidden="true"
          class="breadcrumbs__separator"
        >
          /
        </span>
      </li>
    </ul>
  </nav>
</div>
`;
